### PR TITLE
Added viewer listing to analysis service

### LINKED ISSUE #591: Allow pipeline plugins to use advanved phylogenetic visualization
## Describe your idea for a new feature
What is the feature you would like to see in IRIDA?

Allow a way for pipeline plugins to use the advanced phylogenetic visualization system.

## Describe how you would access this feature?
Where would you interact with this feature?  (New pipeline, new button on the project page, REST API, etc.)

Pipeline plugin developers would have a way to configure their plugins to take advantage of the advanced visualization system.

Users would see the advanced visualization system on the Analysis Results page for specific pipeline plugins.

## Additional information
Add any other context, files, or screenshots about the feature request here.

Logic for displaying advanced visualization handled here:

https://github.com/phac-nml/irida/blob/development/src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisController.java

### PR DESCRIPTION
**Pull request by tom114**
## Description of changes
Added a "viewer" listing to the analysis type service.  This will tell the UI which viewer type to use for different analysis tools.

This also adds a `getAnalysisViewer()` method to `IridaPlugin` so plugin developers can define their own viewer.  Old plugins will not need to be updated as a default is returned if it doesn't implement this method.

## Related issue
Fixes #591 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

File: src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx [hunk 1]
```diff
@@ -75,6 +75,7 @@ export default function Analysis() {
   );
 
   const analysisType = analysisContext.analysisType;
```
[the remark this answers is not in the document]
Comment (tom114) [reply src/main/webapp/resources/js/pages/analysis/components/Analysis.jsx]: Fixed in 2d17398

**Comment by kbessonov1984** [comment]
That is awesome !!! Thank you IRIDA Team! Now it is possible to use built in IRIDA viewers in plugins to render results. I will try to implement `SISTR` serotyping results viewer function in the SISTR plugin so it will interface with the built in viewer functions. I will still need to update a couple of fields in the built-in `SISTR` results viewer and submit PR so more fields will be extracted from the `SISTR` results including `alleles found` field, but it should not be difficult.

**Comment by tom114** [comment]
Glad that should work for you @kbessonov1984 .  Note that this feature will only be available in the newest IRIDA release (yet to be released, expected next month)

**Comment by kbessonov1984** [comment]
Thank you Tom.  I was thinking to continue with this feature testing and further plugin development by pulling this commit and installing it in my docker IRIDA container to try out this new feature. This way I could take advantage of the new IRIDA release features and make new SISTR plugin release.
Kudos for this update.

**Pull request merged**